packages/ddex/ingester/cmd/main.go

```diff
@@ -42,15 +48,188 @@ func main() {
+		}
+		if err := wipeBucket(ingester.S3Client, ingester.CrawledBucket); err != nil {
+			logger.Error("Error creating raw bucket", "err", err)
+		}
+
+		filter := bson.M{}
+		if result, err := ingester.CrawlerCursorColl.DeleteMany(ingester.Ctx, filter); err == nil {
+			log.Printf("Deleted %d crawler_cursor documents\n", result.DeletedCount)
+		} else {
+			logger.Error("Error wiping crawler_cursor collection", "err", err)
+		}
+		if result, err := ingester.DeliveriesColl.DeleteMany(ingester.Ctx, filter); err == nil {
+			log.Printf("Deleted %d deliveries\n", result.DeletedCount)
+		} else {
+			logger.Error("Error wiping deliveries collection", "err", err)
+		}
+		if result, err := ingester.PendingReleasesColl.DeleteMany(ingester.Ctx, filter); err == nil {
+			log.Printf("Deleted %d pending_releases\n", result.DeletedCount)
+		} else {
+			logger.Error("Error wiping pending_releases collection", "err", err)
+		}
+		if result, err := ingester.PublishedReleasesColl.DeleteMany(ingester.Ctx, filter); err == nil {
+			log.Printf("Deleted %d published_releases\n", result.DeletedCount)
+		} else {
+			logger.Error("Error wiping published_releases collection", "err", err)
+		}
+	}
+
+	// Run migration to create artist name index
+	if err := artistutils.CreateArtistNameIndex(ingester.UsersColl, ingester.Ctx); err != nil {
+		log.Fatal(err)
+	}
+
+	// Start the crawler and parser
+	go crawler.CrawlThenParse(ingester, parser.NewParser(ingester).ProcessDelivery)
+
+	// Test the ingester with a delivery if provided
+	if os.Getenv("IS_DEV") == "true" && len(os.Args) > 1 {
+		if err := createBucket(ingester.S3Client, ingester.RawBucket); err != nil {
+			logger.Error("Error creating raw bucket", "err", err)
+		}
+		if err := createBucket(ingester.S3Client, ingester.CrawledBucket); err != nil {
+			logger.Error("Error creating raw bucket", "err", err)
+		}
+		fmt.Printf("Created buckets: %s, %s\n", ingester.RawBucket, ingester.CrawledBucket)
+
+		testDeliveryPath := os.Args[1]
+		testDeliveryURL := uploadTestDelivery(ingester, testDeliveryPath)
+		logger.Info("Uploaded test delivery", "url", testDeliveryURL)
 	}
 
 	<-ctx.Done() // Wait until the context is canceled
 	log.Println("Service stopped")
 }
+
+func createBucket(s3Client *s3.S3, bucket string) error {
+	_, err := s3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucket),
+		CreateBucketConfiguration: &s3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(*s3Client.Config.Region),
+		},
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeBucketAlreadyExists:
+				return fmt.Errorf("bucket name %s already in use", bucket)
+			case s3.ErrCodeBucketAlreadyOwnedByYou:
+				return fmt.Errorf("bucket exists %s and is owned by you", bucket)
+			default:
+				return fmt.Errorf("error creating bucket %s: %v", bucket, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func wipeBucket(s3Client *s3.S3, bucketName string) error {
```

Comment:
wow this is insanely complicated to do... TBH - I wonder if these things should be in the sync.go script or similar. It feels extremely dangerous to have code that deletes entire buckets like this on the main binary

Reply:
i'm not sure if there's a simpler way to wipe with the api but i can look more. and yeah this could be moved to the sync script with a confirmation "--yes" prompt or something. for now it's gated to only run if IS_DEV is true (even if you pass --wipe)

Reply:
yes - i saw the IS_DEV thing, but who knows how that will evolve in the future. I wouldn't expect running with IS_DEV to be true, pointed at a prod s3 bucket to delete the whole thing

Reply:
true true i'll make it double-check that AWS_ENDPOINT is local